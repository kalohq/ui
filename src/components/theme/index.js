/* @flow */
export {default} from './theme';
