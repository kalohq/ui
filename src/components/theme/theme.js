/* @flow */

import tokens from '../../design-tokens/tokens.theme.js';

/**
 * ThemeProvider Kalo Theme
 * 
 * This is the core CSS-in-JS theme that is passed in to the
 * global ThemeProvider. It is mainly composed of our core
 * design tokens.
 */

const theme = {
  space: [0, 8, 16, 32, 64],
  breakpoints: [],
  ...tokens,
};

export default theme;
