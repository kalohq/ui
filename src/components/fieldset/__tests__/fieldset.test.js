/* @flow */
/* eslint-env jest */
import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import {ThemeProvider} from 'emotion-theming';

import theme from '../../theme';
import Input from '../../input';
import Fieldset, {FieldsetHeader} from '../fieldset';
import Field from '../../field';

const defaultProps = {
  inset: false,
};

describe('Input', () => {
  const create = props =>
    shallow(
      <Fieldset {...defaultProps} {...props}>
        <Field label="Your name" required={true}>
          <Input placeholder="Alex" />
        </Field>
      </Fieldset>
    );

  test('should render shallow component ok', () => {
    const element = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Fieldset>
            <Field label="Your name" required={true}>
              <Input placeholder="Alex" />
            </Field>
          </Fieldset>
        </ThemeProvider>
      )
      .toJSON();
    expect(element).toMatchSnapshot();
  });

  test('should render a FieldsetHeader if a legend is passed', () => {
    const element = create({
      legend: 'Your name',
    });
    expect(element.find(FieldsetHeader).length).toBe(1);
  });
});
