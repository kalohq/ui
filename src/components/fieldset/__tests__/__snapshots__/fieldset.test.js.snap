// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render shallow component ok 1`] = `
<fieldset
  className="e102tui80 css-1ne5kij"
  name={undefined}
  onClick={undefined}
  style={Object {}}
>
  <div
    className="e1q9f2xp0 css-1um3mzq"
    onClick={undefined}
    style={Object {}}
  >
    <div
      className="e1q9f2xp0 css-jbik9l"
      style={Object {}}
    >
      <div
        className="e1q9f2xp0 css-1ertpse"
        style={Object {}}
      >
        <label
          className="em7k1990 css-1fy220i"
          color="navy600"
          htmlFor={undefined}
          onClick={undefined}
          size="small"
          style={Object {}}
          title={undefined}
        >
          Your name
          <span
            className="e1q9f2xp3 css-a42xn"
            style={Object {}}
          >
            *
          </span>
        </label>
      </div>
      <div
        className="e6fqpax1 css-15tsz50"
        style={Object {}}
      >
        <input
          className="css-1nv6z25 e6fqpax2"
          defaultValue={undefined}
          disabled={false}
          name="Alex"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Alex"
          readOnly={false}
          size="medium"
          style={
            Object {
              "width": undefined,
            }
          }
          value={undefined}
        />
      </div>
    </div>
  </div>
</fieldset>
`;
